src/kernels/raw/launcher/kernelProcess.node.ts

```diff
@@ -480,20 +483,8 @@ export class KernelProcess implements IKernelProcess {
                     promiseCancellation as Promise<NodeJS.ProcessEnv | undefined>
                 ])
             ]);
-            // Add quotations to arguments if they have a blank space in them.
-            // This will mainly quote paths so that they can run, other arguments shouldn't be quoted or it may cause errors.
             // The first argument is sliced because it is the executable command.
-            const args = this.launchKernelSpec.argv.slice(1).map((a) => {
-                // Some kernel specs (non-python) can have argv as `--connection-file={connection_file}`
-                // The `connection-file` will be quoted when we update it with the real path.
-                if (a.includes('--connection-file')) {
-                    return a;
-                }
-                if (a.includes(' ')) {
-                    return `"${a}"`;
```

Comment:
This was wrong, we were quoting anything that had spaces.

Reply:
Seems correct to me? Why wouldn't we?

Reply:
We should only skip quoting if there was quotes already in the argument. 

What if it looks like this:

```json
"spec": {
        "argv": [
          "C:\\My file\\At\\Spot \\with spaces\\foo.exe",
          "-m",
          "ipykernel_launcher",
          "-f",
          "{connection_file}"
        ],
```

Reply:
If that's the case, then the author of the kernelspec should be quoting that not us.

Reply:
Also, what if me as the author of the kernelspec adds something like
```
"spec": {
        "argv": [
          "C:\\My file\\At\\Spot \\with spaces\\foo.exe",
          "-m",
          "ipykernel_launcher",
          "-f",
          "{connection_file}"
	'some_other_arg="Hello World"'
        ],
```

Now our code comes in and breaks this by adding
```
"spec": {
        "argv": [
          "C:\\My file\\At\\Spot \\with spaces\\foo.exe",
          "-m",
          "ipykernel_launcher",
          "-f",
          "{connection_file}"
	'"some_other_arg="Hello World""'
        ],
```

Reply:
That's why we have to only do this if there aren't already quotes in the argument.

Reply:
I think thats wrong, we'll never get it right, because Jupyter doesn't do anything like that.
Also quoting should be done only for `connection_file` for stuff that we muck around with, and as for other stuff it shoudl be left as is.

Because if we ended up trying to quote stuff that only needs to be quoted, then 100% of the time its unnecessary as Jupyter doesn't do it, meaning it already works and we're not doing anything either.
& if we did have to quote something, then that means we're doing something wrong, when it already works without the necessary quotes in Jupyter.

E.g. how would one end up quoting something from here:
Here's a perfect example, this is a valid OCAML kernelspec.json file
```json
  "argv": [
    "/bin/sh",
    "-c",
    "eval $(opam env --switch=4.13.1 --shell=sh) && /Users/donjayamanne/.opam/4.13.1/bin/ocaml-jupyter-kernel \"$@\"",
    "ocaml-jupyter-kernel",
    "-init", "/Users/donjayamanne/.ocamlinit",
    "--merlin", "/Users/donjayamanne/.opam/4.13.1/bin/ocamlmerlin",
    "--verbosity", "app",
    "--connection-file", "{connection_file}"
  ]
```
I don't think we can some up with any logic that can make this work.
Hence my argument, we shoudl never be quoting these things.

Reply:
If you look at the last part of the 3rd argument, the author (code that genrated the kernelspec) has already added the necessary double quotes, but we come in and break this completely.

Reply:
With the above example, assume we had a kernel spec that didn't have the last part `\"$@\"",` in the 3rd argument then we'd still add the double quotes (based on the rule that `"` doesn't exist, hence lets add it) when its completely unnecessary.

Reply:
So I just created a kernelspec from a venv with spaces in it. By running this:

```
python -m ipykernel install --user --name=VenvWithSpaces
```

Meaning not generating it by hand. 

The kernelspec generated by this did not add quotes:

```
"venvwithspaces": {
      "resource_dir": "C:\\Users\\aku91\\AppData\\Roaming\\jupyter\\kernels\\venvwithspaces",
      "spec": {
        "argv": [
          "D:\\Source\\Testing\\Testing_3\\Venv with Spaces\\Scripts\\python.exe",
          "-m",
          "ipykernel_launcher",
          "-f",
          "{connection_file}"
        ],
        "env": {},
        "display_name": "VenvWithSpaces",
        "language": "python",
        "interrupt_mode": "signal",
        "metadata": {
          "debugger": true
        }
      }
    },
```

Does this kernelspec work with your change? (it works with what we have in stable)

Your ocaml example seems like it would be just fine to me. It would either be this:

```
eval $(opam env --switch=4.13.1 --shell=sh) && /Users/donjayamanne/.opam/4.13.1/bin/ocaml-jupyter-kernel \"$@\"
```

Or if the the user hadn't put quotes anywhere, it would be this:

```
"eval $(opam env --switch=4.13.1 --shell=sh) && /Users/donjayamanne/.opam/4.13.1/bin/ocaml-jupyter-kernel $@"
```

Reply:
If you look at your kernelspec, we never quote the first argument, the first argument is not an argument for the process.
Its the process (command) it self, the rest of the values in `argv` are the arguments passed into the executable `D:\\Source\\Testing\\Testing_3\\Venv with Spaces\\Scripts\\python.exe`
Hence quoting of the first value in `argv` is never required

Reply:
> Your ocaml example seems like it would be just fine to me. It would either be this:

Nope, it was getting changed to this & thus breaks kernel startup.
`"eval $(opam env --switch=4.13.1 --shell=sh) && /Users/donjayamanne/.opam/4.13.1/bin/ocaml-jupyter-kernel \"$@\""`

I.e. we were always adding `"` around any item in `argv` if it contained a space, with the assumption that it needs it.

Reply:
Yes but I'm arguing the bug is that we didn't check for quotes. We shouldn't have added it to that one.

Your argument is we should never add quotes because the kernelspec is already setup to work. Sounds fine now that I realize we're skipping the first argument. That's the one I assume would have spaces in it most of the time.